src/languageTools/DefaultProviders.js

```diff
@@ -46,6 +46,7 @@ define(function (require, exports, module) {
     function CodeHintsProvider(client) {
         this.client = client;
         this.query = "";
+        this.ignoreQuery = ["-", "->", ">", ":", "::", "(", "()", ")", "[", "[]", "]", "{", "{}", "}"];
```

Comment:
Since you have added this list to the defaultprovider, the ignorequery list will be different for different languages. why should this be here?

Reply:
I could have made it specific to PHP but the CodeHintProvider for PHP was using the insertHint from the DefaultProvider, and we needed to know if any query needed to be ignored there as well.

Secondly, this was something that would affect most languages. For PHP I could have stopped at "->" and ":" but then decided to take a more generic approach by including the braces as well, as these things need to be ignored by most languages.

Reply:
IMO, `-` inclusion in ignore list would create problems.  `-` is a valid token character for a lot of languages. HTML5 attributes, CSS property names and values, where we need filtering.